dvc/output/base.py

```diff
@@ -297,8 +297,12 @@ def checkout(self, force=False, progress_callback=None, tag=None):
             progress_callback=progress_callback,
         )
 
-    def remove(self, ignore_remove=False):
-        self.remote.remove(self.path_info)
+    def remove(self, ignore_remove=False, force=True):
+        if self.cache:
+            self.cache.safe_remove(self.path_info, force=force)
+        else:
+            self.remote.remove(self.path_info)
```

Comment:
If file doesn't use cache, then we should also confirm if we want to delete it, right?

Reply:
no, wait, @efiop , if a remote doesn't use `cache` it can't use `external-outputs`, therefore, there's no chance of removing stuff

Reply:
does it make sense to remove this conditional and stay with `self.cache.safe_remove`?

Reply:
Not sure how external outputs are related here. If output doesn't use cache, we are still removing it currently, right? If so, we need to ask user about it, right?

Reply:
Yes, but not _using_ cache is not the same as not _having_ cache.
If you specify `dvc run --outs-no-cache foo` it will _have_ a cache but it won't use it.

Reply:
Discussed privately that `dvc run --outs-no-cache` with external outputs that don't have cache configured will raise this issue.